connection.js

```diff
@@ -20,7 +21,7 @@ function Connection(duplex, opts) {
 
   opts = opts || {}
 
-  var inStream  = generateStream()
+  var inStream  = writeToStream()
```

Comment:
About passing in the original `duplex` stream...I guess I was trying to keep the API compatible with the old `generateStream`, but I think I see what you mean about being able to use that instead of creating the `PassThrough` stream. For a while, I was toggling between `generateStream` and `writeToStream` to ensure the tests were passing with both implementations.